tests/shouldwork/Issues/T2242.hs

```diff
@@ -0,0 +1,16 @@
+module T2242 where
+import System.Environment
+import System.FilePath
+import Clash.Prelude
+topEntity ::
+  HiddenClock System =>
+  Signal System (Unsigned 8) ->
+  Signal System (Unsigned 8)
+topEntity = dflipflop
+{-# NOINLINE topEntity #-}
+
+mainVerilog :: IO ()
+mainVerilog = do
+  [topDir] <- getArgs
+  content <- readFile $ topDir </> show 'topEntity </> "topEntity.v"
+  if '$' `elem` content then error "Error: output contains $ symbol" else pure ()
```

Comment:
You still need to include `mainVHDL` and `mainSystemVerilog` here, since those functions will be needed to test against other HDLs in the test suite

Reply:
mainVHDL and mainSystemVerilog added!